# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/DecodedStream.cpp
# Commit: 1909d94fa9f5
# Full Hash: 1909d94fa9f5d9433f880a69da2f3efc53415757
# Author: alwu <alwu@mozilla.com>
# Date: 2021-04-30 21:45:04
# Description:
#   Bug 1708536 - prevent accessing null 'mData'. r=padenot
#   
#   Depends on D113823
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113855
# ==============================================================================

diff -r 8d038e2b675f -r 1909d94fa9f5 dom/media/mediasink/DecodedStream.cpp
--- a/dom/media/mediasink/DecodedStream.cpp	Fri Apr 30 12:38:05 2021 +0000
+++ b/dom/media/mediasink/DecodedStream.cpp	Fri Apr 30 12:38:06 2021 +0000
@@ -671,7 +671,7 @@
     return;
   }
   mVolume = aVolume;
-  if (mData->mAudioTrack) {
+  if (mData && mData->mAudioTrack) {
     mData->mAudioTrack->SetVolume(static_cast<float>(aVolume));
   }
 }
@@ -686,7 +686,7 @@
     return;
   }
   mPlaybackRate = aPlaybackRate;
-  if (mData->mAudioTrack) {
+  if (mData && mData->mAudioTrack) {
     mData->mAudioTrack->SetPlaybackRate(static_cast<float>(aPlaybackRate));
   }
 }
